### PR TITLE
Clarify extensionKind

### PR DESCRIPTION
**Pull request by egamma**
@bamurtaugh 

I've looked at the change logs from some extensions that have added the `extensionKind` attribute and set it to `["ui"]`. The authors provided the following comments:

>Add ability to use extension without installing on remote

>Sets the extensionKind to ui to allow the extension to be used in remote development environments. This forces the extension to run as a UI extension (on the local machine) rather than a workspace extension (on the remote machine).

This PR attempts to clarify this.

I also suggest to reorder the description of the extensionKind `["ui"]` so that it comes after `["ui", "workspace"]` (I haven't done this so that the word changes show up in the diff).

//CC @alexdima to have a fresh reviewer.

### CODE REVIEW

**Review by alexdima — APPROVED** [review]
:+1:

**Comment by bamurtaugh** [comment]
The changes make sense to me, thanks for opening this @egamma!

> I also suggest to reorder the description of the extensionKind ["ui"] so that it comes after ["ui", "workspace"]

I think this is a good call out - the current ordering might suggest users should start out with `extensionKind ["ui"]` or `extensionKind ["workspace"]` and then only build up to the other two extensionKinds if necessary, but that's not the case.

**Review by bamurtaugh — APPROVED** [review]

File: api/advanced-topics/remote-extensions.md [hunk 1]
```diff
@@ -156,8 +156,8 @@ As of VS Code 1.40, this value is an array which means extensions can specify mo
...
-- `"extensionKind": ["ui", "workspace"]` — Indicates the extension **prefers** to run as a UI extension, but does not have any hard requirements on local assets, devices, or capabilities. When using VS Code, the extension will run in VS Code's local extension host if it exists locally, otherwise will run in VS Code's workspace extension host if it exists there. When using the Codespaces browser-based editor, it will run in the remote extension host always (as no local extension host is available). The old  `"ui"`  value (as a string) maps to this type for backwards compatibility, but is considered deprecated.
+- `"extensionKind": ["ui"]` — Indicates the extension **must** run as a UI extension because it requires access to local assets, devices, or capabilities. Therefore, it can only run in VS Code's local extension host and will not work in the Codespaces browser-based editor (as there is no local extension host available). **Note** Do not use this `extensionKind` when you want to enable that an extension can be used without having to install it on the remote, in this case use the `"extensionKind": ["ui", "workspace"]`.
```
Comment (bamurtaugh) [inline api/advanced-topics/remote-extensions.md:159]: Perhaps the **Note** here could go on a new line or be formatted as a > **Note:** as some other sections; not sure if that'd break up the bullets too much.

**Comment by bamurtaugh** [comment]
@gregvanl We believe this PR is ready for your review, and if you think it looks good, ready to merge as well. Thank you!

**Comment by gregvanl** [comment]
I'll do the reordering before publishing

**Pull request merged**